modules/gapi/samples/face_detection.cpp

```diff
@@ -0,0 +1,764 @@
+        double bboxWidth = x2 - x1;
+        double bboxHeight = y2 - y1;
+        double side = std::max(bboxWidth, bboxHeight);
+        bbox.x1 = static_cast<double>(x1) + (bboxWidth - side) * 0.5;
+        bbox.y1 = static_cast<double>(y1) + (bboxHeight - side) * 0.5;
+        bbox.x2 = bbox.x1 + side;
+        bbox.y2 = bbox.y1 + side;
+        return bbox;
+    }
+};
+
+struct Face {
+    BBox bbox;
+    double score;
+    double regression[NUM_REGRESSIONS];
+    double ptsCoords[2 * NUM_PTS];
+
+    static void applyRegression(std::vector<Face>& faces, bool addOne = false) {
+        for (size_t i = 0; i < faces.size(); ++i) {
+            double bboxWidth =
+                faces[i].bbox.x2 - faces[i].bbox.x1 + static_cast<double>(addOne);
+            double bboxHeight =
+                faces[i].bbox.y2 - faces[i].bbox.y1 + static_cast<double>(addOne);
+            faces[i].bbox.x1 = faces[i].bbox.x1 + static_cast<double>(faces[i].regression[1]) * bboxWidth;
+            faces[i].bbox.y1 = faces[i].bbox.y1 + static_cast<double>(faces[i].regression[0]) * bboxHeight;
+            faces[i].bbox.x2 = faces[i].bbox.x2 + static_cast<double>(faces[i].regression[3]) * bboxWidth;
+            faces[i].bbox.y2 = faces[i].bbox.y2 + static_cast<double>(faces[i].regression[2]) * bboxHeight;
+        }
+    }
+
+    static void bboxes2Squares(std::vector<Face>& faces) {
+        for (size_t i = 0; i < faces.size(); ++i) {
+            faces[i].bbox = faces[i].bbox.getSquare();
+        }
+    }
+
+    static std::vector<Face> runNMS(std::vector<Face>& faces, double threshold,
+                                    bool useMin = false) {
+        std::vector<Face> facesNMS;
+        if (faces.empty()) {
+            return facesNMS;
+        }
+
+        std::sort(faces.begin(), faces.end(), [](const Face& f1, const Face& f2) {
+            return f1.score > f2.score;
+        });
+
+        std::vector<int> indices(faces.size());
+        std::iota(indices.begin(), indices.end(), 0);
+
+        while (indices.size() > 0) {
+            const int idx = indices[0];
+            facesNMS.push_back(faces[idx]);
+            std::vector<int> tmpIndices = indices;
+            indices.clear();
+            for (size_t i = 1; i < tmpIndices.size(); ++i) {
+                int tmpIdx = tmpIndices[i];
+                const double interX1 = std::max(faces[idx].bbox.x1, faces[tmpIdx].bbox.x1);
+                const double interY1 = std::max(faces[idx].bbox.y1, faces[tmpIdx].bbox.y1);
+                const double interX2 = std::min(faces[idx].bbox.x2, faces[tmpIdx].bbox.x2);
+                const double interY2 = std::min(faces[idx].bbox.y2, faces[tmpIdx].bbox.y2);
+
+                const double bboxWidth = std::max(0.0, (interX2 - interX1 + 1));
+                const double bboxHeight = std::max(0.0, (interY2 - interY1 + 1));
+
+                const double interArea = bboxWidth * bboxHeight;
+                // TODO: compute outside the loop
```

Comment:
+1 needs to move `area1`, `area2` from `for`

Reply:
Done. I don't understand point with alignment. Is it better than now? 4 spaces aligned.